sdks/v4-sdk/src/PositionManager.ts

```diff
@@ -252,8 +257,17 @@ export abstract class V4PositionManager {
       planner.addIncrease(options.tokenId, position.liquidity, amount0Max, amount1Max, options.hookData)
     }
 
-    // need to settle both currencies when minting / adding liquidity
-    planner.addSettlePair(position.pool.currency0, position.pool.currency1)
+    // If migrating, we need to settle and sweep both currencies individually
+    if (isMint(options) && options.migrate) {
+      // payer is v4 positiion manager
+      planner.addSettle(position.pool.currency0, false)
+      planner.addSettle(position.pool.currency1, false)
```

Comment:
why can't this just be settlePair?

Reply:
settlePair doesn't have the option of making v4 posm the payer. so you have to do 2 individual settles